src/domain/wallet-invoices/wallet-invoice-validator.ts

```diff
@@ -3,11 +3,7 @@ import { AlreadyPaidError, SelfPaymentError } from "@domain/errors"
 export const WalletInvoiceValidator = (
   walletInvoice: WalletInvoice,
 ): WalletInvoiceValidator => {
-  const validateToSend = ({
-    fromWalletId,
-  }: {
-    fromWalletId: WalletId
-  }): true | ApplicationError => {
+  const validateToSend = (fromWalletId: WalletId): true | ApplicationError => {
```

Comment:
The `ApplicationError` should not be here. `ApplicationError` is scoped to `/app` - this is under domain and should be more specific. Now we have a circular dependency app -> domain -> app... I wonder how this got in here.